swarm/src/connection.rs

```diff
@@ -186,10 +189,17 @@ where
         max_negotiating_inbound_streams: usize,
         idle_timeout: Duration,
     ) -> Self {
-        let initial_protocols = gather_supported_protocols(&handler);
-        if !initial_protocols.is_empty() {
+        let local_supported_protocols = gather_supported_protocols(&handler);
```

Comment:
```suggestion
        let initial_protocols = gather_supported_protocols(&handler);
```

Let's reduce the diff where we can.